src/controllers/higlightController.js

```diff
@@ -0,0 +1,98 @@
+import { config } from 'dotenv';
+import { Article, Highlight } from '../sequelize/models';
+
+config();
+
+/**
+ * @description All the article highlights
+ */
+class Highlights {
+  /**
+     * @description highlight text iof an article
+     * @param {object} req
+     * @param {object} res
+     * @returns {object} error or highlighted object
+     */
+  static async highlightText(req, res) {
+    try {
+      const payload = req.body;
+      const articleSlug = req.params.slug;
+      const userId = req.userData.id;
+
+      const article = await Article.findOne({ where: { slug: articleSlug } });
+
+      if (!article) {
+        return res.status(404).json({
+          error: `Article ${articleSlug} does not exist`
+        });
+      }
+
+      const { body } = article;
+      let highlightedText = 'highlighted';
+      let { comment } = payload;
+      const { startIndex, text } = payload;
+      if (startIndex > body.length) {
+        return res.status(404).json({
+          error: 'Invalid index'
+        });
+      }
+      const textToHighlight = body.slice(startIndex);
+
+      if (!textToHighlight.includes(text)) {
```

Comment:
I recommend you send a startIndex and a stopIndex which would alleviate the need for this check, you'd only check that the stop index does not exceed the body length

Reply:
Ok! makes sense!